File: golang/StatisticsClient.go
package main

import (
	"bytes"
	"encoding/json"
	"github.com/google/uuid"
	"io"
	"log"
	"net/http"
)

type CollectionRequest struct {
	WorkerType string      `json:"workerType"`
	WorkerName string      `json:"workerName"`
	Count      int64       `json:"count"`
	Ids        []uuid.UUID `json:"ids"`
}

func SendStatistics(workerType string, workerName string, count int64, ids []uuid.UUID) {
	url := ConfigInstance.StatisticsCollectorURL
	req := CollectionRequest{
		WorkerType: workerType,
		WorkerName: workerName,
		Count:      count,
		Ids:        ids,
	}

	data, err := json.Marshal(req)
	if err != nil {
		log.Printf("json.Marshal failed: %s", err)
		return
	}

	resp, err := http.Post(url+"/collect", "application/json", bytes.NewBuffer(data))
	if err != nil {
		log.Printf("http.Post failed: %s", err)
		return
	}

	defer func(Body io.ReadCloser) {
		_ = Body.Close()
	}(resp.Body)

	log.Println("Statistics sent to the statistics collector", resp.Status)
}

File: python/statistics_client.py
import os

from dotenv import load_dotenv
import requests

load_dotenv()

enable_statistics = os.getenv("ENABLE_STATISTICS", "false").lower() == "true"
collector_url = os.getenv("STATISTICS_COLLECTOR_URL", "http://localhost:5000")


def load_worker_name():
    print("Loading worker name")
    if enable_statistics:
        response = requests.post(f"{collector_url}/worker-count")


def send_statistics(data):
    if enable_statistics:
        response = requests.post(f"{collector_url}/collect", json=data)
        print(response.json())
    else:
        print("Statistics disabled")
